Panic: runtime error. Unsure if resources request/limits are causing it

**Describe the bug**
Getting a panic runtime error sometimes.

**To Reproduce**
I played with resources and limits and maybe this is happening with the resources and limits. But unsure since the pod does not seem to be hitting any limits.

We also applied resources and limits as follows:

```
resources:
  requests:
    cpu: 50m
    memory: 75Mi
  limits:
     cpu: 500m
     memory: 512Mi
```

We are running across many different applications. We use name and semver image strategies and an internal docker repo. The error does seem to be happening right after log lines about a project with an image list of 120+ image tags but we have seen the error after other image lists as well.


**Expected behavior**
Be able to use resources requests

**Additional context**
Using install.yaml manifest with patches applied to add image secrets for our registry and to change interval to 1m0s in args.


 
**Version**
0.9.4

**Logs**
```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x17224ac]


goroutine 6951 [running]:

github.com/argoproj-labs/argocd-image-updater/pkg/tag.ImageTagList.Add(0xc000f67980, 0xc0009c4f40, 0x0)

/src/argocd-image-updater/pkg/tag/tag.go:98 +0x5c

github.com/argoproj-labs/argocd-image-updater/pkg/registry.(*RegistryEndpoint).GetTags.func1(0xc000d1ba40, 0xc0011a5880, 0x20a0f80, 0xc00000e5b8, 0xc000f10580, 0xc00000e5c0, 0xc000584120, 0xc000f10570, 0xc0003d08c0, 0xc0005ac6f0, ...)

/src/argocd-image-updater/pkg/registry/registry.go:157 +0x2a0

created by github.com/argoproj-labs/argocd-image-updater/pkg/registry.(*RegistryEndpoint).GetTags

/src/argocd-image-updater/pkg/registry/registry.go:121 +0x95e
```

--- a/pkg/registry/registry.go
+++ b/pkg/registry/registry.go
@@ -28,7 +28,6 @@ const (
 // GetTags returns a list of available tags for the given image
 func (endpoint *RegistryEndpoint) GetTags(img *image.ContainerImage, regClient RegistryClient, vc *image.VersionConstraint) (*tag.ImageTagList, error) {
 	var tagList *tag.ImageTagList = tag.NewImageTagList()
-	var imgTag *tag.ImageTag
 	var err error
 
 	// Some registries have a default namespace that is used when the image name
@@ -76,7 +75,7 @@ func (endpoint *RegistryEndpoint) GetTags(img *image.ContainerImage, regClient R
 			} else if endpoint.TagListSort == SortLatestLast {
 				ts = i
 			}
-			imgTag = tag.NewImageTag(tagStr, time.Unix(int64(ts), 0), "")
+			imgTag := tag.NewImageTag(tagStr, time.Unix(int64(ts), 0), "")
 			tagList.Add(imgTag)
 		}
 		return tagList, nil
@@ -97,7 +96,7 @@ func (endpoint *RegistryEndpoint) GetTags(img *image.ContainerImage, regClient R
 		// an error, we treat it as a cache miss and just go ahead to invalidate
 		// the entry.
 		if vc.SortMode.IsCacheable() {
-			imgTag, err = endpoint.Cache.GetTag(nameInRegistry, tagStr)
+			imgTag, err := endpoint.Cache.GetTag(nameInRegistry, tagStr)
 			if err != nil {
 				log.Warnf("invalid entry for %s:%s in cache, invalidating.", nameInRegistry, imgTag.TagName)
 			} else if imgTag != nil {
@@ -153,7 +152,7 @@ func (endpoint *RegistryEndpoint) GetTags(img *image.ContainerImage, regClient R
 			}
 
 			log.Tracef("Found date %s", ti.CreatedAt.String())
-
+			var imgTag *tag.ImageTag
 			if vc.SortMode == image.VersionSortDigest {
 				imgTag = tag.NewImageTag(tagStr, ti.CreatedAt, fmt.Sprintf("sha256:%x", ti.Digest))
 			} else {

